pandas/core/frame.py

```diff
@@ -8419,7 +8419,9 @@ def _get_data(axis_matters):
                 raise NotImplementedError(msg)
             return data
 
-        if numeric_only is not None and axis in [0, 1]:
+        is_numeric = all(b.is_numeric for b in self._mgr.blocks)
```

Comment:
do the added tests hit this sufficiently (e.g. some blocks numeric some blocks not)

Reply:
no, but that is another issue #34520